jax/_src/numpy/polynomials/chebyshev.py

```diff
@@ -0,0 +1,679 @@
+def _zseries_to_cseries(zs):
+  n = (zs.size + 1)//2
+  c = zs[n-1:]
+  c = c.at[1:n].multiply(2)
+  return c
+
+@jit
+def _zseries_mul(z1, z2):
+  return jnp.convolve(z1, z2)
+
+@jit
+def _zseries_div(z1, z2):
+  lc1 = len(z1)
+  lc2 = len(z2)
+
+  if lc2 == 1:
+    z1 /= z2
+    return z1, z1[:1]*0
+  elif lc1 < lc2:
+    return z1[:1]*0, z1
+  else:
+    dlen = lc1 - lc2
+    scl = z2[0]
+    z2 /= scl
+    quo = jnp.zeros(dlen + 1, dtype=z1.dtype)
+
+    i = 0
+    j = dlen
+
+    def cond_fun(val):
+      return val[0] < val[1]
+
+    def body_fun(val):
+      i = val[0]
+      j = val[1]
+      quo = val[2]
+      z1 = val[3]
+      r = z1[i]
+      quo = quo.at[i].set(r)
+      quo = quo.at[j].set(r)
+      tmp = r*z2
+      z1 = _chebadd(jnp.roll(z1, -i), -tmp)
+      z1 = _chebadd(jnp.roll(z1, i-j), -tmp)
+      z1 = jnp.roll(z1, j)
+      i += 1
+      j -= 1
+      return(i, j, quo, z1)
+
+    i, j, quo, z1 = lax.while_loop(cond_fun, body_fun, (i, j, quo, z1))
+    r = z1[i]
+    quo = quo.at[i].set(r)
+    tmp = r*z2
+    z1 = _chebadd(jnp.roll(z1, -i), -tmp)
+    quo /= scl
+    rem = jnp.roll(z1, -1)[:lc2-2]
+    return quo, rem
+
+@jit
+def _poly2cheb(pol):
+  [pol] = pu._as_series([pol])
+  pol, = _promote_dtypes_inexact(pol)
+  deg = len(pol) - 1
+  res = jnp.zeros(deg+1, dtype=pol.dtype)
+  pol = jnp.flip(pol)
+
+  def body_fun(res, p):
+    prd = jnp.zeros(len(res), dtype=res.dtype)
+    prd = prd.at[1].set(res[0])
+    if len(res) > 1:
+      tmp = res[1:-1]/2
+      prd = prd.at[2:].set(tmp)
+      prd = prd.at[0:-2].add(tmp)
+    prd = prd.at[0].add(p)
+    return prd, p
+
+  res, _ = lax.scan(body_fun, res, pol)
+  return res
+
+def poly2cheb(pol, trim=True):
+  if trim:
+    return _poly2cheb(pu.trimseq(pol))
+  else:
+    return _poly2cheb(pol)
+
+@jit
+def _cheb2poly(c):
+  [c] = pu._as_series([c])
+  c, = _promote_dtypes_inexact(c)
+  n = len(c)
+  if n < 3:
+    return c
+  else:
+    c0 = jnp.zeros(len(c), dtype=c.dtype)
+    c0 = c0.at[0].set(c[-2])
+    c1 = jnp.zeros(len(c), dtype=c.dtype)
+    c1 = c1.at[0].set(c[-1])
+
+    def body_fun(i, val):
+      v0 = _chebsub(c[-i-2], val[1])
+      v1 = jnp.zeros(len(val[1]), dtype=val[1].dtype)
+      v1 = v1.at[1:].set(val[1][:-1])
+      v1 = _chebadd(val[0], 2*v1)
+      return(v0, v1)
+
+    c0, c1 = lax.fori_loop(-(n-1),-1, body_fun, (c0, c1))
+
+    tmp = jnp.zeros(len(c1), dtype=c1.dtype)
+    c1 = tmp.at[1:].set(c1[:-1])
+
+    return _chebadd(c0, c1)
+
+def cheb2poly(c, trim=True):
+  if trim:
+    return _cheb2poly(pu.trimseq(c))
+  else:
+    return _cheb2poly(c)
+
+@jit
+def _chebline(off, scl):
+  return jnp.array([off, scl])#, dtype=jnp.promote_types(off, scl))
+
+def chebline(off, scl, trim=True):
+  if trim:
+    return pu.trimseq(_chebline(off, scl))
+  else:
+    return _chebline(off, scl)
+
+@jit
+def chebfromroots(roots):
+  if len(roots) == 0:
+    return jnp.ones(1, dtype=roots.dtype)
+  else:
+    [roots] = pu._as_series([roots])
+    roots.sort()
```

Comment:
```suggestion
    roots = roots.sort()
```
`sort` is not an inplace operation.